sklearn/tree/_criterion.pyx

```diff
@@ -347,8 +347,9 @@ cdef class ClassificationCriterion(Criterion):
 
         # Reset to pos=start
         self.reset()
+        return 0
```

Comment:
We should return the output of reset here, no? How can the Cython execption detection work otherwise? If we always return zero we basically tell Cython that it should never check for the presence of an exception to be raised.